plugins/modules/azure_rm_webapp_accessrestriction.py

```diff
@@ -0,0 +1,364 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2021 Ross Bender (@l3ender)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from inspect import signature
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: azure_rm_webapp_accessrestriction
+version_added: "1.9.0"
+short_description: Manage web app network access restrictions
+description:
+    - Add, remove, or update network access restrictions for a web app.
+options:
+    name:
+        description:
+            - Name of the web app.
+        required: true
+        type: str
+    resource_group:
+        description:
+            - Resource group of the web app.
+        required: true
+        type: str
+    state:
+        description:
+            - State of the access restrictions. Use C(present) to create or update and C(absent) to delete.
+        type: str
+        default: present
+        choices:
+            - absent
+            - present
+    ip_security_restrictions:
+        description:
+            - The web app's HTTP access restrictions.
+        type: list
+        suboptions:
+            name:
+                description:
+                    - Name of the access restriction.
+                type: str
+            description:
+                description:
+                    - Description of the access restriction.
+                type: str
+            action:
+                description:
+                    - Traffic action for the access restriction.
+                type: str
+                default: Allow
+                choices:
+                    - Allow
+                    - Deny
+            priority:
+                description:
+                    - Numerical priority of the access restriction.
+                type: int
+                required: true
+            ip_address:
+                description:
+                    - IPv4 address (with subnet mask) of the access restriction.
+                type: str
+                required: true
+    scm_ip_security_restrictions:
+        description:
+            - The web app's SCM access restrictions. If C(scm_ip_security_restrictions_use_main) is set to C(true), the SCM restrictions will be configured but not used.
+        type: list
```

Comment:
Argument 'scm_ip_security_restrictions' in argument_spec specifies elements as dict,but elements is not documented
```suggestion
        type: list
        elements: dict
```